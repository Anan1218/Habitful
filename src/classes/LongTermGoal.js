import React from "react";
import {
    StyleSheet,
    View,
    Alert,
    Modal,
    TouchableHighlight,
    TouchableWithoutFeedback,
} from "react-native";
import "react-native-gesture-handler";
import Grid from "../styles/Grid";

import { Icon, Block, Text, Button, Input } from "galio-framework";
import { removeGoal, updateGoal } from "../dbFunctions/GoalFunctions";
export default class LongTermGoal extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            modalVisible: false,
            title: this.props.title,
            newGoalTitle: this.props.title,
        };
    }

    render() {
        return (
            // <Block card={true} style={styles.card}>

            <View style={[Grid.col, styles.LongTermGoal]}>
                <View style={Grid.row}>
                    <View style={Grid.col}>
                        <View style={[Grid.row, styles.headingText]}>
                            <Text p>{this.state.title}</Text>
                        </View>
                        <View style={Grid.row}>
                            <View style={[Grid.row, styles.infoText]}>
                                <Text muted>
                                    {this.props.habits} running habits
                                </Text>
                            </View>
                            <View style={[Grid.row, styles.infoText]}>
                                <Text muted>
                                    {this.props.TDs} To-Dos completed
                                </Text>
                            </View>
                        </View>
                    </View>

                    <View style={[Grid.col]}>
                        <Button
                            onlyIcon
                            icon="edit"
                            iconFamily="antdesign"
                            iconSize={20}
                            color="warning"
                            iconColor="#fff"
                            style={{ width: 35, height: 35 }}
                            onPress={() => {
                                this.setState({ modalVisible: true });
                            }}
                        ></Button>
                    </View>
                </View>

                {/*
                <Modal
                    animationType="slide"
                    transparent={true}
                    visible={this.state.modalVisible}
                    onRequestClose={() => {
                        this.setState({ modalVisible: false });
                    }}
                >
                    <TouchableWithoutFeedback
                        onPress={() => {
                            this.setState({ modalVisible: false });
                        }}
                    >
                        <View
                            style={[Grid.row, Grid.justifyCenter, { flex: 1 }]}
                        >
                            <View
                                style={[
                                    Grid.col,
                                    Grid.justifyCenter,
                                    styles.modalView,
                                    Grid.alignStretch,
                                ]}
                            >
                                <View
                                    style={[
                                        Grid.col,
                                        Grid.justifyCenter,
                                        Grid.alignStretch,
                                    ]}
                                >
                                    <Button
                                        onlyIcon
                                        icon="delete"
                                        iconFamily="antdesign"
                                        iconSize={20}
                                        color="red"
                                        iconColor="#fff"
                                        style={{ width: 35, height: 35 }}
                                        onPress={() => {
                                            this.props.deleteGoal(
                                                this.props.id
                                            );
                                        }}
                                    ></Button>
                                    <Input
                                        placeholder="e.g. Maintain a healthy lifestyle"
                                        onChangeText={(text) =>
                                            this.setState({
                                                newGoalTitle: text,
                                            })
                                        }
                                        value={this.state.newGoalTitle}
                                    />
                                </View>
                                <View style={[Grid.row, Grid.justifyCenter]}>
                                    <Button
                                        size="small"
                                        onPress={() => {
                                            updateGoal(
                                                this.state.title,
                                                this.state.newGoalTitle
                                            );
                                            this.setState({
                                                title: this.state.newGoalTitle,
                                            });
                                            this.setState({
                                                modalVisible: false,
                                            });
                                        }}
                                    >
                                        <Text>Change title</Text>
                                    </Button>
                                </View>
                            </View>
                        </View>
                    </TouchableWithoutFeedback>
                </Modal>
                                      */}
            </View>

            // </Block>
        );
    }
}
const styles = StyleSheet.create({
    card: {
        margin: 7,
        backgroundColor: "black",
    },
    LongTermGoal: {
        margin: 10,
    },
    headingText: {
        paddingRight: 19,
        paddingLeft: 14,
    },
    infoText: {
        paddingRight: 19,
        paddingLeft: 14,
    },
});
