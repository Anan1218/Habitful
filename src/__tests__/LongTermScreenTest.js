import 'react-native';
import React from 'react';

import LongTermGoal from "../classes/LongTermGoal";

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <LongTermGoal />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
