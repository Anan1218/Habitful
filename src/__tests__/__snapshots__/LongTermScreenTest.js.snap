// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "flex-start",
        "alignItems": "stretch",
        "display": "flex",
        "flexDirection": "column",
      },
      Object {
        "margin": 10,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignContent": "flex-start",
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      style={
        Object {
          "alignContent": "flex-start",
          "alignItems": "stretch",
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignContent": "flex-start",
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "flex-start",
            },
            Object {
              "paddingLeft": 14,
              "paddingRight": 19,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              false,
              false,
              false,
              false,
              false,
              Object {
                "fontSize": 20,
              },
              false,
              undefined,
              0,
              null,
              false,
              false,
              undefined,
              null,
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "alignContent": "flex-start",
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignContent": "flex-start",
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "flex-start",
              },
              Object {
                "paddingLeft": 14,
                "paddingRight": 19,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                false,
                false,
                false,
                false,
                false,
                false,
                Object {
                  "color": "#9FA5AA",
                },
                undefined,
                0,
                null,
                false,
                false,
                undefined,
                null,
              ]
            }
          >
             running habits
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignContent": "flex-start",
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "flex-start",
              },
              Object {
                "paddingLeft": 14,
                "paddingRight": 19,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                false,
                false,
                false,
                false,
                false,
                false,
                Object {
                  "color": "#9FA5AA",
                },
                undefined,
                0,
                null,
                false,
                false,
                undefined,
                null,
              ]
            }
          >
             To-Dos completed
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignContent": "flex-start",
            "alignItems": "stretch",
            "display": "flex",
            "flexDirection": "column",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FF9C09",
            "borderRadius": 20,
            "borderWidth": 0,
            "elevation": 1,
            "height": 35,
            "justifyContent": "center",
            "opacity": 1,
            "shadowColor": "#FF9C09",
            "shadowOffset": Object {
              "height": 4,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "shadowRadius": 10,
            "width": 35,
            "zIndex": 2,
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignContent": "flex-start",
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          },
          Object {
            "justifyContent": "center",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignContent": "flex-start",
              "alignItems": "stretch",
              "display": "flex",
              "flexDirection": "column",
            },
            Object {
              "justifyContent": "center",
            },
            undefined,
            Object {
              "alignItems": "stretch",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignContent": "flex-start",
                "alignItems": "stretch",
                "display": "flex",
                "flexDirection": "column",
              },
              Object {
                "justifyContent": "center",
              },
              Object {
                "alignItems": "stretch",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "marginVertical": 8,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#808080",
                    "borderRadius": 8,
                    "borderWidth": 0.8,
                    "height": 44,
                    "paddingHorizontal": 16,
                    "width": "100%",
                  },
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  null,
                  false,
                  false,
                  null,
                ]
              }
            >
              <TextInput
                allowFontScaling={true}
                keyboardType="default"
                onChangeText={[Function]}
                onRef={null}
                placeholder="e.g. Maintain a healthy lifestyle"
                placeholderTextColor={null}
                rejectResponderTermination={true}
                secureTextEntry={false}
                style={
                  Array [
                    Object {
                      "flex": 1,
                    },
                    false,
                    Object {
                      "color": "#808080",
                      "fontSize": 14,
                      "textDecorationColor": "transparent",
                      "textShadowColor": "transparent",
                    },
                    null,
                  ]
                }
                underlineColorAndroid="transparent"
              />
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignContent": "flex-start",
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "flex-start",
              },
              Object {
                "justifyContent": "center",
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#B23AFC",
                "borderRadius": 3,
                "elevation": 1,
                "height": 44,
                "justifyContent": "center",
                "opacity": 1,
                "shadowColor": "#B23AFC",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.8,
                "shadowRadius": 10,
                "width": 375,
                "zIndex": 2,
              }
            }
          >
            <Text
              style={
                Array [
                  false,
                  false,
                  false,
                  false,
                  false,
                  false,
                  false,
                  undefined,
                  0,
                  null,
                  false,
                  false,
                  undefined,
                  null,
                ]
              }
            >
              Change title
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
